ravenframework/Optimizers/BayesianOptimizer.py

```diff
@@ -0,0 +1,758 @@
+
+    acquisitionFunction = InputData.parameterInputFactory('Acquisition', strictMode=True,
+                                                          printPriority=106,
+                                                          descr=r"""A required node for specifying
+                                                          the details about the acquisition function
+                                                          used in the policy of Bayesian Optimization.""")
+
+    # Pulling specs for each acquisition function option
+    for option in acqFactory.knownTypes():
+      subSpecs = acqFactory.returnClass(option).getInputSpecification()
+      acquisitionFunction.addSub(subSpecs)
+    specs.addSub(acquisitionFunction)
+
+    # Model selection
+    modelSelect = InputData.parameterInputFactory('ModelSelection', strictMode=False,
+                                                  printPriority=106,
+                                                  descr=r"""An optional node allowing the user to specify the details of model selection.
+                                                  For example, the manner in which hyperparameters are selected for the GPR model.""")
+    modelSelect.addSub(InputData.parameterInputFactory('Duration', contentType=InputTypes.IntegerType,
+                                                       descr=r"""Number of iterations between each reselection of the model. Default is 1"""))
+    modelSelect.addSub(InputData.parameterInputFactory('Method', contentType=InputTypes.makeEnumType("Method", "MethodType", ['External', 'Internal', 'Average']),
+                                                       descr=r"""Determines methodology for selecting the model. This methodology is applied
+                                                       after every duration length.
+                                                       \begin{itemize}
+                                                       \item External, uses whatever method the model has internal to itself
+                                                       \item Internal, selects the MAP point of the model using slsqp
+                                                       \item Average, Approximate marginalization over model space
+                                                       \end{itemize}. Default is External"""))
+    specs.addSub(modelSelect)
+
+    # Convergence
+    conv = InputData.parameterInputFactory('convergence', strictMode=True,
+                                           printPriority=108,
+                                           descr=r"""a node containing the desired convergence criteria for the optimization algorithm.
+                                           Note that convergence is met when any one of the convergence criteria is met. If no convergence
+                                           criteria are given, then the defaults are used.""")
```

Comment:
I am a bit confused by this comment since this is a parent node. The child nodes have default values of 5 for persistence and 1e-8 for the acquisition criteria.

Reply:
The description says "then the defaults are used", I thought some defaults values will be provided.